hugr-core/src/std_extensions/collections.rs

```diff
@@ -134,10 +151,19 @@ impl ListOp {
         let e = Type::new_var_use(0, TypeBound::Any);
         let l = self.list_type(list_type_def, 0);
         match self {
-            pop => self
-                .list_polytype(vec![l.clone()], vec![l.clone(), e.clone()])
-                .into(),
+            pop => self.list_polytype(vec![l.clone()], vec![l, e]).into(),
             push => self.list_polytype(vec![l.clone(), e], vec![l]).into(),
+            concat => self
+                .list_polytype(vec![l.clone(), l.clone()], vec![l])
+                .into(),
+            lookup => self.list_polytype(vec![l, USIZE_T], vec![e]).into(),
+            length => self.list_polytype(vec![l], vec![USIZE_T]).into(),
+            repeat => self.list_polytype(vec![l.clone(), USIZE_T], vec![l]).into(),
+            count => self.list_polytype(vec![l, e], vec![USIZE_T]).into(),
```

Comment:
How are the elements compared when counted? We don't have a constraint system yet that could ensure that the elements have a well-defined equality operation. Until then we could make this a dynamic thing of course. Or we could pass in an equality predicate?